file.go

```diff
@@ -160,14 +160,15 @@ func (fi *File) Flush() error {
 		return err
 	}
 
-	defer fd.Close()
-
-	return fd.Flush()
+	// Close calls fd.Flush() and will do a full sync since it was opened with
+	// Sync=true. So, there is not need to call fd.Flush()
+	return fd.Close()
 }
 
 func (fi *File) Sync() error {
```

Comment:
Who even calls this? Can't we just remove it?

Reply:
Maybe,  but since this function is exported it could be called externally or used to satisfy some interface somewhere.  I did not want to introduce any potential breakage in this PR.

Reply:
Fair enough. Usually `go-ipfs` is the main consumer here but we should check more broadly (@Stebalien ?) and clean in a follow-up PR.

Reply:
Go-ipfs is pretty much the only consumer.

Reply:
Removed.  Cannot find any use of it.